### ISSUE TITLE
Trouble with Valet 3.0 alpha due to Homebrew PHP formula's default `www` pools

### ISSUE THREAD

**Issue by nicoverbruggen** [laravel/valet#1205]
Hey @mattstauffer and @NasirNobin,

First off, my apologies for this wall of text 😅...

I've been trying out the new changes in Valet 3.0's alpha, and I have started work on integrating site isolation into PHP Monitor. I absolutely love that you guys were able to add this to Valet — it's the one thing I've been missing all this time. I'm grateful you were able to add this to Valet. ❤️

I did run into a particular issue on my system that presents itself when multiple conflicting PHP services are running, and I wanted to report this — or at least see if you are having the same issue(s). I'm not entirely positive if this is a "oops, it's happening on my system" kinda issue or if it's going to be a common issue.

I believe that the default installation of a PHP formula creates a `www` pool configuration file in `/opt/homebrew/etc/php/{version}/php-fpm.d/www.conf`. (Correct me if I'm wrong — I found these files on my system, at least. I think this _may_ be a more recent addition, I'm not sure. Can you confirm that you have these files on your system as well?)

Anyway, my issue seems to stem from these `www` pool configuration files.

Since multiple PHP-FPM instances might be running if isolated sites are configured, it is possible that due to this default file, the various `www` pools are created, all trying to bind to the default port 9000. (Each PHP-FPM instance spins up all pools set up in `/opt/homebrew/etc/php/{version}/php-fpm.d`. Note that on Intel systems your Homebrew folder might be in a different spot.)

Only one PHP version's `www` pool will "win" port 9000 in this case, and all others will silently fail in the background. As a result, in `sudo brew services list` these services will now report themselves as failing. Note that Valet should, at this point in time, remain functional, since its `valet-fpm.conf` pools are not affected... (yet).

I think it might be prudent to disable these default `www` pools now that Valet uses multiple PHP versions, because... due to Homebrew thinking that these PHP-FPM `www` pools conflict, it is possible that some of the commands that Valet needs to execute, fail (ones that involve `brew services`). 

As far as I can tell, on my system, the various PHP-FPM processes did not spin up the respective `valet` pools if one of the `www` pools failed to bind to port 9000. This doesn't always seem to happen, so I'm guessing the order in which the pools are started is random or perhaps the pools are started in parallel? The point is, sometimes the Valet socket won't work because the PHP-FPM pool created for Valet won't start for some reason (again, I'm guessing it's due to the port conflict).

So, basically... This default `www.conf` file might also be one of the erratic causes of socket trouble in the past with Valet. I have a theory that sometimes, running `brew services restart php@{version}` will not work if one such conflict arises, which means that not only does the `www` pool not work, but the Valet pool might also fail to start, resulting in a potential 502 for all sites using the socket that, for all intents and purposes, should've worked if the `www.conf` file were not present or pointing at a valid, non-conflicting port. In the past, this meant that *all* sites Valet should serve would fail, but now this means only *some* isolated sites might fail to be served now that multiple PHP-FPM processes are responsible for each serving various sites (if isolation is enabled).

This conflict can occur if you have *any kind of software running that actively uses port 9000 at the time of you starting Valet's services*. (Port 9000 is a common pick. I've used some JS-based server stuff that would bind to that port. I didn't think about it then, but I remember having sporadic issues with Valet then as well.)

Renaming the default file `www.conf` to something else that won't start conflicting pools would probably fix that issue; alternatively each of those `www.conf` files could also bind to different ports. 

I suspect that most users won't have a need for these `www` pools anyway, so in order to avoid a bunch of issues disabling them might be a good idea once Valet starts. To do so, one would just need to rename the `www.conf` files (assuming they exist). I purged my `www.conf` files and I've had zero issues since, switching between PHP versions and isolating sites. (Love the new feature, like I said!)

TL;DR:

* Did you also find the `www.conf` files in Homebrew's `/etc/php/{version}/php-fpm.d` directory?
* If so, did these all point towards port 9000? (Thus causing conflicts?)
* Did you have issues with services reporting errors when running `sudo brew services list`?
* Is it a good idea to perhaps disable these `www.conf` files by default when Valet starts? This could prevent socket issues.

P.S. I did also notice that the new `valet secured` command (#1203) lists dotfiles as "secured sites". I'm guessing this is an oversight and I wanted to report that as well, but didn't want to make a bunch of issues at once.

Cheers!

**Comment by NasirNobin** [comment]
> Did you also find the www.conf files in Homebrew's /etc/php/{version}/php-fpm.d directory?

I was able to reproduce this issue. This is happening due to removing [these few lines](https://github.com/laravel/valet/commit/eefc06b07f8ee57186ce898cacaaf33f9504d679#diff-d8e7b9c98bc6a045292f784be0d98860c26b463a1dd29f36256fed1d2ad62752L89) most probably. 

I'll try to dig a bit more into this and will try to maybe open a pull request fixing it.

**Comment by mattstauffer** [comment]
@NasirNobin Whoops... Thank you so much for catching that mistake I introduced!

@nicoverbruggen Thanks for noticing the bug and writing it up.

**Comment by mattstauffer** [comment]
@nicoverbruggen could you open a new issue for the `secured`  concern?

**Comment by nicoverbruggen** [comment]
@mattstauffer Done, I've created #1210!
